arch/x86/core/thread.c

```diff
@@ -167,6 +167,15 @@ void _x86_swap_update_page_tables(struct k_thread *incoming,
 	/* TODO: if either thread defines different memory domains, efficiently
 	 * switch between them
 	 */
+	if (incoming->mem_domain_info.mem_domain !=
+	   outgoing->mem_domain_info.mem_domain){
+
+		/* There maybe inconsistencies in the page states.
```

Comment:
What do you mean by "maybe inconsistencies"
Taking down the outgoing memory domain configuration is required on context switch.